albumentations/augmentations/functional.py

```diff
@@ -1008,35 +1008,23 @@ def grid_distortion(
     """
     height, width = img.shape[:2]
 
-    x_step = width // num_steps
-    xx = np.zeros(width, np.float32)
+    x_split = np.array_split(range(width), num_steps)
     prev = 0
-    for idx, x in enumerate(range(0, width, x_step)):
-        start = x
-        end = x + x_step
-        if end > width:
-            end = width
-            cur = width
-        else:
-            cur = prev + x_step * xsteps[idx]
-
-        xx[start:end] = np.linspace(prev, cur, end - start)
+    for i, chunk in enumerate(x_split):
+        cur = prev + len(chunk) * xsteps[i]
+        split = np.linspace(prev, cur, len(chunk))
+        x_split[i] = split
         prev = cur
+    xx = np.concatenate(x_split)
```

Comment:
I think we can create one function for this case, something like `split_axis`. After it this code will look like this:
```
x_split = split_axis(width, num_steps, xsteps)
y_split = split_axis(height, num_steps, ysteps)
```